modules/1plusXRtdProvider.js

```diff
@@ -0,0 +1,211 @@
+ */
+const extractConfig = (moduleConfig, reqBidsConfigObj) => {
+  // CustomerId
+  const customerId = deepAccess(moduleConfig, 'params.customerId');
+  if (!customerId) {
+    throw new Error('Missing parameter customerId in moduleConfig');
+  }
+  // Timeout
+  const tempTimeout = deepAccess(moduleConfig, 'params.timeout');
+  const timeout = isNumber(tempTimeout) && tempTimeout > 300 ? tempTimeout : 1000;
+
+  // Bidders
+  const biddersTemp = deepAccess(moduleConfig, 'params.bidders');
+  if (!isArray(biddersTemp) || !biddersTemp.length) {
+    throw new Error('Missing parameter bidders in moduleConfig');
+  }
+
+  const adUnitBidders = reqBidsConfigObj.adUnits
+    .flatMap(({ bids }) => bids.map(({ bidder }) => bidder))
+    .filter((e, i, a) => a.indexOf(e) === i);
+  if (!isArray(adUnitBidders) || !adUnitBidders.length) {
+    throw new Error('Missing parameter bidders in bidRequestConfig');
+  }
+
+  const bidders = biddersTemp.filter(
+    bidder =>
+      SUPPORTED_BIDDERS.includes(bidder) && adUnitBidders.includes(bidder)
+  );
+  if (!bidders.length) {
+    throw new Error('No supported bidder found in config parameter bidders');
+  }
+
+  return { customerId, timeout, bidders };
+}
+
+/**
+ * Gets the URL of Profile Api from which targeting data will be fetched
+ * @param {Object} config
+ * @param {string} config.customerId
+ * @returns {string} URL to access 1plusX Profile API
+ */
+const getPapiUrl = ({ customerId }) => {
+  // https://[yourClientId].profiles.tagger.opecloud.com/[VERSION]/targeting?url=
+  const currentUrl = encodeURIComponent(window.location.href);
+  const papiUrl = `https://${customerId}.profiles.tagger.opecloud.com/${PAPI_VERSION}/targeting?url=${currentUrl}`;
+  return papiUrl;
+}
+
+/**
+ * Fetches targeting data. It contains the audience segments & the contextual topics
+ * @param {string} papiUrl URL of profile API
+ * @returns {Promise} Promise object resolving with data fetched from Profile API
+ */
+const getTargetingDataFromPapi = (papiUrl) => {
+  return new Promise((resolve, reject) => {
+    const requestOptions = {
+      customHeaders: {
+        'Accept': 'application/json'
+      }
+    }
+    const callbacks = {
+      success(responseText, response) {
+        resolve(JSON.parse(response.response));
+      },
+      error(error) {
+        reject(error);
+      }
+    };
+    ajax(papiUrl, callbacks, null, requestOptions)
+  })
+}
+
+/**
+ * Prepares the update for the ORTB2 object
+ * @param {Object} targetingData Targeting data fetched from Profile API
+ * @param {string[]} segments Represents the audience segments of the user
+ * @param {string[]} topics Represents the topics of the page
+ * @returns {Object} Object describing the updates to make on bidder configs
+ */
+export const buildOrtb2Updates = ({ segments = [], topics = [] }) => {
+  const userData = {
+    name: '1plusX.com',
+    segment: segments.map((segmentId) => ({ id: segmentId }))
+  };
```

Comment:
Here is the element I will add to `ortb2.user.data` containing the audience segments